### PR TITLE
OAK-10798: Introduce functions to clean nodes based on regular expression

### PR DESCRIPTION
**Pull request by Joscorbe**
This adds a few functions to the [oak-mongo.js](https://github.com/apache/jackrabbit-oak/blob/ad7e429b02f724a05679ca0e96ab2c45d0e2a6e7/oak-run/src/main/js/oak-mongo.js) in [oak-run](https://github.com/apache/jackrabbit-oak/tree/trunk/oak-run) to perform some low-level operations.

### CODE REVIEW

File: oak-run/src/main/js/oak-mongo.js [hunk 1]
```diff
@@ -329,6 +329,59 @@ var oak = (function(global){
...
     };
 
+    /**
+     * Helper method to find nodes based on Regular Expression.
+     *
+     * @memberof oak
+     * @method regexFind
+     * @param {string} pattern the pattern to match the nodes.
+     */
+	api.regexFind = function(pattern) {
```
Comment (stefan-egli) [inline oak-run/src/main/js/oak-mongo.js]: indenting is not beautified (tabs instead of spaces)

File: oak-run/src/main/js/oak-mongo.js [hunk 2]
```diff
@@ -329,6 +329,59 @@ var oak = (function(global){
...
+
+    /**
+     * Remove the complete subtree of all the nodes matching a regex pattern.
+     * Use regexFind to find the nodes that match the pattern prior deletion.
+     *
+     * @memberof oak
+     * @method removeDescendantsMatching
+     * @param {string} pattern the pattern to match the nodes to be removed.
+     */
+    api.removeDescendantsMatching = function(pattern) {
```
Comment (stefan-egli) [inline oak-run/src/main/js/oak-mongo.js]:
```suggestion
    api.removeDescendantsAndSelfMatching = function(pattern) {
```

naming : should we use a similar name to `removeDescendantsAndSelf` to stress that `self` will also be deleted?

**Review by stefan-egli — COMMENTED** [review]
Generally looks good, just two detail aspects noted. Also noticed that the original `removeDescendantsAndSelf` doesn't stress anything about such low level operations potentially breaking certain constraints - so I don't think we need to add that now neither.

File: oak-run/src/main/js/oak-mongo.js (hunk 2 above)
Comment (Joscorbe) [reply oak-run/src/main/js/oak-mongo.js]: Done

**Review by rishabhdaim — APPROVED** [review]

**Review by stefan-egli — APPROVED** [review]

**Pull request merged**